pilot/user/atlas/container.py

```diff
@@ -50,12 +50,23 @@ def get_payload_proxy(proxy_outfile_name, voms_role='atlas'):
             return False
 
         proxy_contents = res['userProxy']
-
+        
     except Exception as e:
         logger.error("Get proxy from panda server failed: %s, %s" % (e, traceback.format_exc()))
         return False
 
-    return write_file(proxy_outfile_name, proxy_contents, mute=False)
+    res = False;
+    try:
+        # pre-create empty proxy file with secure permissions. Prepare it for write_file() which can not 
+        # set file permission mode, it will writes to the existing file with correct permissions.
+        f=os.open(proxy_outfile_name, os.O_WRONLY|os.O_CREAT|os.O_TRUNC, 0o600)
```

Comment:
Please add a try-statement around this in case of failure to prevent a crash

Reply:
"try" is already there, before comments :)

Reply:
this try-except block will catch all open() exceptions,
write_file() is also in the try block now, since it doesn't catch exception from open()  itself.

Reply:
sorry, please ignore that..